salt/_modules/azurearm_network.py

```diff
@@ -2793,20 +2801,150 @@ def virtual_network_gateway_connection_create_or_update(name, resource_group, vi
 
     netconn = __utils__['azurearm.get_client']('network', **kwargs)
 
-    # Get a VirtualNetWorkGateway Object using the name of the Virtual Network 
-    # Gateway that the connection will originate from.
-    vnetgw = virtual_network_gateway_get( 
-	name=virtual_network_gateway1,
-        resource_group=resource_group,
-        **kwargs
-    )
+    # Converts the Resource ID of virtual_network_gateway1 into a VirtualNetworkGateway Object.
+    # This endpoint will be where the connection originates.
```

Comment:
Maybe something like this:
```
        if not is_valid_resource_id(virtual_network_gateway):
            log.error(
                'Invalid Resource ID was specified as virtual_network_gateway!'
            )
            return False

        vnetgw1_parsed_id = parse_resource_id(virtual_network_gateway)

        try:
            vnetgw1_rg = vnetgw1_parsed_id['resource_group']
            vnetgw1_name = vnetgw1_parsed_id['resource_name']
        except KeyError as exc:
            log.error(
                'Invalid Resource ID was specified as virtual_network_gateway! (%s)',
                exc
            )
            return False

        vnetgw1 = virtual_network_gateway_get(
            name=vnetgw1_name,
            resource_group=vnetgw1_rg,
            **kwargs
        )

        if 'error' in vnetgw1:
            log.error(
                'Unable to find the resource specified in virtual_network_gateway!'
            )
            return False
        else:
            virtual_network_gateway = {'id': str(vnetgw1['id'])}

    # Check the Resource ID path of virtual_network_gateway2
    # We can't guarantee the validity of the object, so we hope you have the path correct...
    if kwargs.get('virtual_network_gateway2') and not is_valid_resource_id(kwargs['virtual_network_gateway2']):
            log.error(
                'Invalid Resource ID was specified as virtual_network_gateway2!'
            )
            return False

    # Check the Resource ID path of local_network_gateway2
    # We can't guarantee the validity of the object, so we hope you have the path correct...
    if kwargs.get('local_network_gateway2') and not is_valid_resource_id(kwargs['local_network_gateway2']):
            log.error(
                'Invalid Resource ID was specified as local_network_gateway2!'
            )
            return False
```

Reply:
For virtual_network_gateway2, we definitely can't guarantee that we'll be able to access it with the same subscription credentials we're passing to create the connection. I don't know about local_network_gateway2, so we can either handle it like the VNET gateway we're working on and validate it or like the "other" VNET gateway depending on how it's handled. Do a little research and let me know what you think. Oh... and I would drop the "1" from the virtual_network_gateway1 parameter name since the connection is a SubResource of the gateway. It'll be the one where we're initiating the connection. (I'm speculating here... feel free to push back)